### PR TITLE
Read csv records with different delimeter

### LINKED ISSUE #229: add "read csv records with different delimeter"
example should show reading csv with delimeter different than ','

cc https://github.com/brson/rust-cookbook/issues/212

See [CONTRIBUTING.md](https://github.com/brson/rust-cookbook/blob/master/CONTRIBUTING.md) for details.

### PR DESCRIPTION
**Pull request by alisha17**
fixes #229

### CODE REVIEW

File: src/encoding.md [hunk 1]
```diff
@@ -344,6 +345,44 @@ fn run() -> Result<()> {
...
+Reads CSV records with delimiter other than ','.
+
+```rust
+extern crate csv;
+
+use std::error::Error;
+use csv::ReaderBuilder;
+use std::process;
+
+fn run() -> Result<(), Box<Error>> {
```
Comment (budziq) [inline src/encoding.md]: please do not use Box<Error>. We prefer error_chain for the sake of conformity

File: src/encoding.md [hunk 2]
```diff
@@ -344,6 +345,44 @@ fn run() -> Result<()> {
...
+    let mut reader = ReaderBuilder::new().delimiter(b'-').from_reader(data.as_bytes());
+    for result in reader.records() {
+        let record = result?;
+        println!("{:?}", record);
+    }
+
+    Ok(())
+}
+
+fn main() {
```
Comment (budziq) [inline src/encoding.md]: PLease use error_chain's `quick_main`

File: src/encoding.md [hunk 3]
```diff
@@ -344,6 +345,44 @@ fn run() -> Result<()> {
...
+
+```rust
+extern crate csv;
+
+use std::error::Error;
+use csv::ReaderBuilder;
+use std::process;
+
+fn run() -> Result<(), Box<Error>> {
+    let data = "\
```
Comment (budziq) [inline src/encoding.md]: is the extra linebreak needed here?

File: src/encoding.md [hunk 4]
```diff
@@ -344,6 +345,44 @@ fn run() -> Result<()> {
...
+        name-place-id
+        Mark-Melbourne-46
+        Ashley-Zurich-92";
+
+    let mut reader = ReaderBuilder::new().delimiter(b'-').from_reader(data.as_bytes());
```
Comment (budziq) [inline src/encoding.md]: please mention and link the `delimiter` method

File: src/encoding.md [hunk 5]
```diff
@@ -344,6 +345,44 @@ fn run() -> Result<()> {
...
+    for result in reader.records() {
+        let record = result?;
```
Comment (budziq) [inline src/encoding.md]: do we need this extra line? why not use `?` inside the `println!`?

**Comment by brson** [comment]
Looks good but needs a rebase to deal with the merge conflict.

**Comment by brson** [comment]
I rebased and merged manually. thanks @alisha17

**Pull request closed without merging**